Common/Securities/Forex/Forex.cs

```diff
@@ -101,19 +102,111 @@ public Forex(Symbol symbol, SecurityExchangeHours exchangeHours, Cash quoteCurre
         public string BaseCurrencySymbol { get; private set; }
 
         /// <summary>
-        /// Decomposes the specified currency pair into a base and quote currency provided as out parameters
+        /// Decomposes the specified currency pair into a base and quote currency provided as out parameters.
+        /// Requires symbols in Currencies.CurrencySymbols dictionary to make accurate splits, important for crypto-currency symbols.
         /// </summary>
         /// <param name="currencyPair">The input currency pair to be decomposed, for example, "EURUSD"</param>
         /// <param name="baseCurrency">The output base currency</param>
         /// <param name="quoteCurrency">The output quote currency</param>
         public static void DecomposeCurrencyPair(string currencyPair, out string baseCurrency, out string quoteCurrency)
         {
-            if (currencyPair == null || currencyPair.Length != 6)
+            if (currencyPair == null || currencyPair.Length < 6 || currencyPair.Length > Currencies.MaxCharactersPerCurrencyPair)
             {
-                throw new ArgumentException("Currency pairs must be exactly 6 characters: " + currencyPair);
+                throw new ArgumentException($"Currency pairs must not be null, length minimum of 6 and maximum of {Currencies.MaxCharactersPerCurrencyPair}. Problematic pair: {currencyPair}");
```

Comment:
Moved it, now this function and one other residues in Util.cs.
Made it so that default SecurityType is Base (so no need to change current functionality anywhere).